src/components/app/create/CustomTagSelector/TagLabelValueSelector.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useState, useEffect } from 'react'
+import { OptionType, TagType } from '../../../app/types'
+import { PopupMenu } from '../../../common'
+import { ValidationRules } from '../validationRules'
+import { ReactComponent as ErrorCross } from '../../../../assets/icons/ic-close.svg'
+import { ReactComponent as Info } from '../../../../assets/icons/ic-info-outlined.svg'
+
+export default function TagLabelValueSelector({
+    selectedTagIndex,
+    tagData,
+    setTagData,
+    tagOptions,
+    isRequired,
+    type,
+}: {
+    selectedTagIndex: number
+    tagData: TagType
+    setTagData: (index: number, tagData: TagType) => void
+    tagOptions?: OptionType[]
+    isRequired?: boolean
+    type?: string
+}) {
+    const [selectedValue, setSelectedValue] = useState<string>('')
+    const [isPopupOpen, togglePopup] = useState<boolean>(false)
+
+    const validationRules = new ValidationRules()
+
+    useEffect(() => {
+        const _tagData = { ...tagData }
+        setSelectedValue(_tagData[type])
+    }, [selectedTagIndex, tagData, type])
+
+    const handleOnBlur = (e) => {
+        if (
+            !e.relatedTarget ||
+            !e.relatedTarget.classList.value ||
+            e.relatedTarget.classList.value.indexOf(`tag-${selectedTagIndex}-class`) === -1
+        ) {
+            const _tagData = { ...tagData }
+            _tagData[type] = selectedValue
+            if (type === 'key') {
+                _tagData.isInvalidKey = selectedValue
+                    ? !validationRules.propagateTagKey(selectedValue).isValid
+                    : _tagData.value !== ''
+            } else {
+                if (selectedValue) {
+                    _tagData.isInvalidValue = !validationRules.propagateTagValue(selectedValue).isValid
+                    _tagData.isInvalidKey = !_tagData.key || _tagData.isInvalidKey
+                } else {
+                    _tagData.isInvalidValue = false
+                    _tagData.isInvalidKey = !_tagData.key ? false : _tagData.isInvalidKey
+                }
```

Comment:
Here can we use if...else-if statement simply?

Reply:
done